src/otx/algorithms/detection/adapters/mmdet/models/detectors/custom_maskrcnn_tile_optimized.py

```diff
@@ -31,7 +31,7 @@ def __init__(self):
             ConvModule(192, 256, 3, padding=1, act_cfg=dict(type="ReLU")),
             nn.MaxPool2d(kernel_size=3, stride=2),
         )
-        self.avgpool = torch.nn.AdaptiveAvgPool2d((6, 6))
+        self.avgpool = torch.nn.AvgPool2d(5, stride=2)
```

Comment:
It's just a question. Is it ok to use `AvgPool2d` with constant value? Isn't there any possibility to make error with different input shape?

Reply:
> It's just a question. Is it ok to use `AvgPool2d` with constant value? Isn't there any possibility to make error with different input shape?

Good review, it could be risky in the configurable input size scenario.

Reply:
For me, adaptive pool seems more appropriate

Reply:
Adaptive pooling doesn't work when I ran export. Instead of pooling it to 6 x 6, it outputs 7 x 7 in ONNX.

Reply:
I don't get it, Do you mean ov model will return different feature size with pytorch model?

Reply:
> I don't get it, Do you mean ov model will return different feature size with pytorch model?

No, the export failed due to an issue from ONNX tracing, not from OV.

```
TracerWarning: Converting a tensor to a Python integer might cause the trace to be incorrect. 
We can't record the data flow of Python values, so this value will be treated as a constant in the future. 
This means that the trace might not generalize to other inputs!
  k = [int(size[i] / output_size[i]) for i in range(0, len(size))]
```

Reply:
So fixes for ir_scale to 1.0 raises error during onnx export due to dimension mismatch right?
If so, still there is one remaining question.
Static average pooling is okay for various input size? Have you conducted experiments for various image input size?

Reply:
> So fixes for ir_scale to 1.0 raises error during onnx export due to dimension mismatch right? If so, still there is one remaining question. Static average pooling is okay for various input size? Have you conducted experiments for various image input size?

I agree with you that adaptive pooling is more appropriate, but essentially there's a discrepancy of how PyTorch computes adaptive parameters vs. mmdeploy. I think what mmdeploy implemented is wrong.

![image](https://github.com/openvinotoolkit/training_extensions/assets/13793254/a92aa1ff-4a2f-47ed-86e6-3c9b88056706)

Reply:
I changed it to global average pooling and conducted an experiment on Coliform. The difference is negligible

|                                      | f-measure      |   
|-----------------------------------------|--------------------|
| Global Average Pooling       | 0.5855670103092783 |   
| Adaptive Average Pooling(6,6) | 0.5854993160054718 |

Reply:
Yes, mmdeploy's implementation seems wrong. I'm a little concerned about effect to model performance with other dataset but changing it in some way is inevitable. Could you add some comments to say the history of this in short? and I think flatten can be removed if gloabl_avg_pool(1,1) is used. How about to remove it?

Reply:
@eunwoosh I changed it to GAP as follows:

![image](https://github.com/openvinotoolkit/training_extensions/assets/13793254/7ad5da85-e3d4-486f-8e16-7826ebfffd8a)